src/services/api/peopleApi.ts

```diff
@@ -0,0 +1,62 @@
+import { apiClient } from '@src/services/api/apiClient';
+import type {
+  MultiPagePeopleApiRequest,
+  PeopleApiResponse,
+  Person,
+} from '@src/types/apiTypes';
+import { getApiPageNumber } from '@src/helpers/getApiPageNumber';
+import { FetchBaseQueryError } from '@reduxjs/toolkit/query';
+import { ITEMS_PER_PAGE } from '@src/constants/constants';
+
+const peopleApi = apiClient.injectEndpoints({
+  endpoints: (builder) => ({
+    getPerson: builder.query<Person, number>({
+      query: (id) => {
+        const method = 'GET';
+        const url = `people/${String(id)}`;
+
+        return { url, method };
+      },
+    }),
+    getMultiPagePeople: builder.query<
+      PeopleApiResponse,
+      MultiPagePeopleApiRequest
+    >({
+      async queryFn(args, _api, _extraOptions, fetchWithBQ) {
+        const { itemsPerPage, searchValue, page } = args;
+        const apiPageNumber = getApiPageNumber(page ?? 1, itemsPerPage);
+
+        const firstResult = await fetchWithBQ(
+          `people/?page=${apiPageNumber}${
+            searchValue && searchValue.length
+              ? '&search=' + encodeURIComponent(searchValue)
+              : ''
+          }`,
+        );
+
+        if (firstResult.error)
+          return { error: firstResult.error as FetchBaseQueryError };
+
+        const data = firstResult.data as PeopleApiResponse;
+        let restQueriesCount = itemsPerPage / ITEMS_PER_PAGE.DEFAULT - 1;
+        let nextPageURL = data.next;
+
+        while (nextPageURL && restQueriesCount > 0) {

```

Comment:
`while (nextPageURL && restQueriesCount) {`

Reply:
Maybe strict comparison is better for numbers. Negative valuses can be truthy.